Makefile

```diff
@@ -70,22 +70,31 @@ help: ## Display this help.
 .PHONY: generate
 generate: ## Generate boilerplate DeepCopy methods, manifests, and Go client
 	# Use uid and gid of current user to avoid mismatched permissions
-	iidfile=$$(mktemp /tmp/iid-XXXXXX) && \
+	set -e ; \
+	iidfile=$$(mktemp /tmp/iid-XXXXXX) ; \
 	docker build \
 		--build-arg USER_ID=$(shell id -u $(USER)) \
 		--build-arg GROUP_ID=$(shell id -g $(USER)) \
 		--build-arg CONTROLLER_TOOLS_VERSION=$(CONTROLLER_TOOLS_VERSION) \
 		--build-arg CODE_GENERATOR_VERSION=$(CODE_GENERATOR_VERSION) \
 		--file neonvm/hack/Dockerfile.generate \
-		--iidfile $$iidfile . && \
+		--iidfile $$iidfile . ; \
+	volumes=('--volume' "$$PWD:/go/src/github.com/neondatabase/autoscaling") ; \
+	if [ -f .git ]; then \
+		gitdir="$$(git rev-parse --git-common-dir)" ; \
+		volumes+=('--volume') ; \
+		volumes+=("$$gitdir:$$gitdir") ; \
+	fi ; \
+	set -x ; \
 	docker run --rm \
-		--volume $$PWD:/go/src/github.com/neondatabase/autoscaling \
+		"$${volumes[@]}" \
+		--volume "$$volumes" \
```

Comment:
Not sure if we need this line as we already have the line above.
It adds an empty `--volume` argument to the docker command:

```
docker run --rm --volume /home/runner/work/autoscaling/autoscaling:/go/src/github.com/neondatabase/autoscaling --volume --volume --workdir /go/src/github.com/neondatabase/autoscaling --user 1001:127 sha256:8f356c0be7cebf483fd219ae1e189b8ea5579ad666c613bee24ca19f9f10e7ef ./neonvm/hack/generate.sh
```

Ref https://github.com/neondatabase/autoscaling/actions/runs/10606700736/job/29397913790?pr=1053#step:3:203


```suggestion
```